aws/resource_aws_appconfig_environment.go

```diff
@@ -0,0 +1,240 @@
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"monitor": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alarm_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"alarm_role_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentName := aws.String(d.Get("name").(string))
+	environmentDescription := aws.String(d.Get("description").(string))
+
+	var monitorList []*appconfig.Monitor
+	if monitors := d.Get("monitor").(*schema.Set).List(); len(monitors) > 0 {
+		monitorList = convertMapToMonitors(monitors)
+	}
+
+	input := &appconfig.CreateEnvironmentInput{
+		ApplicationId: applicationID,
+		Name:          environmentName,
+		Description:   environmentDescription,
+		Monitors:      monitorList,
+		Tags:          keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	env, err := conn.CreateEnvironment(input)
+	if err != nil {
+		return fmt.Errorf("Creating AppConfig Environment failed: %s", err)
+	}
+	log.Printf("[DEBUG] AppConfig Environment created: %s", env)
+
+	d.SetId(aws.StringValue(env.Id))
+
+	return resourceAwsAppconfigEnvironmentRead(d, meta)
+}
+
+func resourceAwsAppconfigEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	input := &appconfig.GetEnvironmentInput{
+		ApplicationId: aws.String(d.Get("application_id").(string)),
+		EnvironmentId: aws.String(d.Id()),
+	}
+
+	output, err := conn.GetEnvironment(input)
+
+	if isAWSErr(err, appconfig.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Appconfig Environment (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting AppConfig Environment (%s): %s", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting AppConfig Environment (%s): empty response", d.Id())
+	}
+
+	appARN := arn.ARN{
+		AccountID: meta.(*AWSClient).accountid,
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("application/%s/environment/%s", aws.StringValue(output.ApplicationId), aws.StringValue(output.Id)),
+		Service:   "appconfig",
+	}.String()
+
+	d.Set("arn", appARN)
+	d.Set("name", output.Name)
+	d.Set("description", output.Description)
+	d.Set("monitor", convertMonitorsToMap(output.Monitors))
```

Comment:
I'm confused at the monitor --> monitors conversion

Reply:
See https://github.com/figma/terraform-provider-aws/pull/79#discussion_r594688082